src/DGtal/shapes/MeshVoxelizer.ih

```diff
@@ -0,0 +1,241 @@
+inline
+void
+DGtal::MeshVoxelizer<TDigitalSet, Separation>::voxelizeTriangle(DigitalSet &outputSet,
+                                                                const PointR3& A,
+                                                                const PointR3& B,
+                                                                const PointR3& C,
+                                                                const VectorR3& n,
+                                                                const std::pair<PointR3, PointR3>& bbox)
+{
+  OrientationFunctor orientationFunctor;
+
+  //geometric predicate
+  PredicateFromOrientationFunctor2<OrientationFunctor> pointPredicate( orientationFunctor );
+
+  // foreach intersection target
+  for(unsigned int i(0); i < myIntersectionTarget().size(); i++)
+  {
+    // 2D projection of A ; B ; C
+    PointR2 AA = myIntersectionTarget.project(i, A);
+    PointR2 BB = myIntersectionTarget.project(i, B);
+    PointR2 CC = myIntersectionTarget.project(i, C);
+
+    // check orientation
+    if(! pointPredicate(AA, BB, CC))
+      std::swap(AA, CC);
+
+    // traverse all voxel of current face bounding box
+    PointZ3 v = bbox.first;
+    for(; v[1] <= bbox.second[1]; v[1]++)
+      for(v[0] = bbox.first[0]; v[0] <= bbox.second[0]; v[0]++)
+        for(v[2] = bbox.first[2]; v[2] <= bbox.second[2]; v[2]++)
+        {
+          auto target = myIntersectionTarget(i);
+          // check if points are on different side
+          target.myFirst += v;
+          target.mySecond += v;
+
+          VectorR3 a2myFirst = A - target.myFirst;
+          VectorR3 a2mySecond = A - target.mySecond;
+
+          VectorR3 w = target.mySecond - target.myFirst;
+          double den = n.dot(w);
+
+          bool isSameSide = den == 0 // target on plane
+            || ( a2myFirst.dot(n) * a2mySecond.dot(n) > 0 ); // target on one side
+
+          // if target is on the same side -> skip current iteration
+          if( isSameSide )
+            continue;
+
+          PointR2 pp = myIntersectionTarget.project(i, v);
+
+          // check if current voxel projection is inside ABC projection
+          if(pointIsInside2DTriangle(AA, BB, CC, pp) != OUTSIDE)
+            outputSet.insert(v);
+        }
+  }
+}
+
+// ---------------------------------------------------------
+template <typename TDigitalSet, int Separation>
+template <typename MeshPoint>
+inline
+void
+DGtal::MeshVoxelizer<TDigitalSet,Separation>::voxelize(DigitalSet &outputSet,
+                                                       const MeshPoint &a,
+                                                       const MeshPoint &b,
+                                                       const MeshPoint &c,
+                                                       const double scaleFactor)
+{
+  std::pair<PointR3, PointR3> bbox;
+  VectorR3 n, e1, e2;
+  PointR3 A, B, C;
+
+  //Scaling + casting to PointR3
+  A = a*scaleFactor;
+  B = b*scaleFactor;
+  C = c*scaleFactor;
+
```

Comment:
perhaps add an assert to check if the points are in the digital set domain ?

Reply:
yeah thanks.

Reply:
Ok I've slightly changed the interface: I first check that the voxel is inside the domain or not before the insert. Hence, no more issue with the domain but if a voxel is outside the set, it is simply skipped.
Fine for you ?

Reply:
yes fine